### ISSUE TITLE
Avoiding instrumented calls producing report output during test setup

### ISSUE THREAD

**Issue by nbarrett** [serenity-bdd/serenity-core#669]
I use some instrumented tasks to perform some setup / reset of environment state before quite a few of my scenarios. This works great from a functional point of view but (predictably), I see those steps appearing on the first few lines of the report. For example in the screenshot below, the first highlighted steps are produced by my setup  in the `@Before` annotation and the second section is the actual `@Test`. 

![image](https://cloud.githubusercontent.com/assets/626941/22425133/8fac8b72-e6f1-11e6-8868-c146a28fb428.png)

So my question is: How do I turn off instrumentation until I want it? Or can I clear steps from the event bus at a specific point before my 'real' test begins? I could think of a good place to put this call but I don't know what I need to call. Obviously I only want to clear stuff from the current test case, not anything before that.

**Comment by nbarrett** [comment]
Hi John - any thoughts on this one please?

**Comment by nbarrett** [comment]
Hi John - any thoughts on this before I do some hack in my branch like this?

```
    public static <T extends Performable> T instrumentOptionally(Class<T> stepClass, Object... parameters) {
        boolean REPORT_ON_STEPS = true;
        if (REPORT_ON_STEPS) {
            return instrumented(stepClass, parameters);
        } else {
            try {
                return stepClass.getConstructor(Attribute.class).newInstance(parameters);
            } catch (Exception e) {
                throw new IllegalStateException("Unable to create step for " + stepClass);
            }
        }

    }
```

**Comment by nbarrett** [comment]
Hi John - any thoughts on this before I do some hack in my branch like this?
```
    public static <T extends Performable> T instrumentOptionally(Class<T> stepClass, Object... parameters) {
        if (REPORT_ON_STEPS) {
            return instrumented(stepClass, parameters);
        } else {
            try {
                return stepClass.getConstructor(Attribute.class).newInstance(parameters);
            } catch (Exception e) {
                throw new IllegalStateException("Unable to create step for " + stepClass);
            }
        }

    }
```

where `REPORT_ON_STEPS` is a boolean  value that I turn on and off during the test, depending on whether I want report output

**Comment by wakaleo** [comment]
Something like that could work, the trick would be to know how to trigger the boolean flag.

**Comment by wakaleo** [comment]
Implemented with the CanBeSilent interface in 2.0.12
